tests/unit/oidc/test_models.py

```diff
@@ -132,3 +133,26 @@ def test_github_provider_verifies(self, monkeypatch):
         }
         assert provider.verify_claims(signed_claims=signed_claims)
         assert len(noop_check.calls) == len(verifiable_claims)
+
+    @pytest.mark.parametrize(
+        ("claim", "valid"),
+        [
+            ("foo/bar/.github/workflows/baz.yml@", True),
```

Comment:
Isn't this going to be something like `foo/bar/.github/workflows/baz.yml@refs/pull/6/merge`?

Reply:
Whoops, you're absolutely right. I need to tweak it to ensure that there's data following the `@`, including at least one `/` (to prevent the phony filename problem).

Reply:
[798a29f](https://github.com/pypa/warehouse/pull/11263/commits/798a29ffbbb3e716ffadfee430eadd88235d4c0b)

Reply:
As far as I can tell, I'm not sure this addresses the issue of reusable workflows, since it checks that what's in that string is from the same repo as the main repo we're validating.

I think we can't do it properly without a new db field because the information is independant.

Also, what happens if there's an `@` in the branch/tag name ? I'm starting to think that it may not be possible to parse this without injecting knowledge. Is the `ref` part passed elsewhere ?

Reply:
We've decided (short term) not to support reusable workflows, but long term we will want to support them.

Reply:
Yeah, that long term goal is captured in #11096.